docs/manual/src/futures.md

```diff
@@ -36,11 +36,16 @@ if __name__ == '__main__':
 
 This code uses `asyncio` to drive the future to completion, while our exposed function is used with `await`.
 
-In Rust `Future` terminology this means the foreign bindings supply the "executor" - think event-loop, or async runtime. In this example it's `asyncio`. There's no requirement for a Rust event loop.
-
 There are [some great API docs](https://docs.rs/uniffi_core/latest/uniffi_core/ffi/rustfuture/index.html) on the implementation that are well worth a read.
 
-See the [foreign-executor fixture](https://github.com/mozilla/uniffi-rs/tree/main/fixtures/foreign-executor) for more implementation details.
+## Foreign executors
+
+UniFFI futures are normally usually driven by a foreign executor - think event-loop, or async runtime.
+In the example above it's `asyncio`.
+There's no requirement for a Rust executor like tokio, although you can manually start one and use it alongside the foreign executor.
+
+Rust code can also use foreign executors directly to schedule tasks, a typical use case is scheduling work in a background thread.
+For more info see the [ForeignExecutor API](https://docs.rs/uniffi_core/latest/uniffi_core/ffi/foreignexecutor/struct.ForeignExecutor.html) and the [futures example crate](https://github.com/mozilla/uniffi-rs/tree/main/examples/futures).
```

Comment:
The `ForeignExecutor` docs.rs API will get the python/kotlin/swift examples from the struct docstring once this gets merged to main.